main.py

```diff
@@ -759,20 +762,47 @@ def handle_link_up(self, event):
 
     # Possibly replace this with interruptions?
     @listen_to(
-        '.*.switch.interface.(link_up|link_down|created|deleted)',
-        pool='dynamic_single'
+        '.*.switch.interface.(link_up|link_down|created|deleted)'
     )
     def on_interface_link_change(self, event: KytosEvent):
         """
-        Handler for interface link_up and link_down events
+        Handler for interface link_up and link_down events.
+
+        To avoid multiple database updated (link flap):
+        Every interface is identfied and processed in parallel.
+        Once an interface event is received a time is started.
+        While time is running self.intf_events will be updated.
+        After time has passed last received event will be processed.
         """
-        with self._lock:
-            _, _, event_type = event.name.rpartition('.')
-            iface = event.content.get("interface")
-            if event_type in ('link_up', 'created'):
-                self.handle_interface_link_up(iface)
-            elif event_type in ('link_down', 'deleted'):
-                self.handle_interface_link_down(iface)
+        iface = event.content.get("interface")
+        with self.lock_interfaces[iface.id]:
+            _now = event.timestamp
+            # Return out of order events
+            if (
+                iface.id in self.intf_events
+                and self.intf_events[iface.id]["event"].timestamp > _now
+            ):
+                return
+            self.intf_events[iface.id].update({"event": event})
+            if self.intf_events[iface.id].get("time_lock"):
+                return
+            self.intf_events[iface.id].update({"time_lock": True})
```

Comment:
We can do better instead of just using a bool here. To improve future debug-ability, since at this point in time here we now that the thread handler events are being processed in order, you could use `now()` -> `{"last_acquired": now()}`, and then later on you just pop the `"last_acquired"` key when it's no longer acquired, replacing the if `if self.intf_events[iface.id].get("time_lock")` with `if "last_acquired" not in self.intf_events[iface.id]` then early returning as you're doing. In summary, just trying to use a type a bit more richer than the bool one in this case. If we ever need to further debug this in the future, then we can rely on "last_acquired" for further correlation, even though it's acting like a boolean, but with richer (more meaningful) data, see what I mean?.

Reply:
Now using `last_acquired` to check active delay